mscore/linearize.cpp

```diff
@@ -0,0 +1,343 @@
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2002-2014 Werner Schweer
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2
+//  as published by the Free Software Foundation and appearing in
+//  the file LICENCE.GPL
+//=============================================================================
+
+/**
+ File handling: loading and saving.
+ */
+
+#include "config.h"
+#include "globals.h"
+#include "musescore.h"
+#include "scoreview.h"
+#include "exportmidi.h"
+#include "libmscore/xml.h"
+#include "libmscore/element.h"
+#include "libmscore/note.h"
+#include "libmscore/chord.h"
+#include "libmscore/rest.h"
+#include "libmscore/sig.h"
+#include "libmscore/clef.h"
+#include "libmscore/key.h"
+#include "instrdialog.h"
+#include "libmscore/score.h"
+#include "libmscore/page.h"
+#include "libmscore/dynamic.h"
+#include "file.h"
+#include "libmscore/style.h"
+#include "libmscore/tempo.h"
+#include "libmscore/select.h"
+#include "libmscore/staff.h"
+#include "libmscore/part.h"
+#include "libmscore/utils.h"
+#include "libmscore/barline.h"
+#include "libmscore/slur.h"
+#include "libmscore/hairpin.h"
+#include "libmscore/ottava.h"
+#include "libmscore/textline.h"
+#include "libmscore/pedal.h"
+#include "libmscore/trill.h"
+#include "libmscore/volta.h"
+#include "libmscore/timesig.h"
+#include "libmscore/box.h"
+#include "libmscore/excerpt.h"
+#include "libmscore/system.h"
+#include "libmscore/tuplet.h"
+#include "libmscore/keysig.h"
+#include "libmscore/measure.h"
+#include "libmscore/undo.h"
+#include "libmscore/repeatlist.h"
+#include "libmscore/beam.h"
+#include "libmscore/stafftype.h"
+#include "libmscore/revisions.h"
+#include "libmscore/lyrics.h"
+#include "libmscore/segment.h"
+#include "libmscore/tempotext.h"
+#include "libmscore/sym.h"
+#include "libmscore/image.h"
+#include "synthesizer/msynthesizer.h"
+#include "svggenerator.h"
+#include "libmscore/tiemap.h"
+#include "libmscore/tie.h"
+#include "libmscore/measurebase.h"
+
+#include "importmidi/importmidi_instrument.h"
+
+#include "libmscore/chordlist.h"
+#include "libmscore/mscore.h"
+#include "thirdparty/qzip/qzipreader_p.h"
+#include "thirdparty/qzip/qzipwriter_p.h"
+
+
+namespace Ms {
+
+   // Influenced strongly by appendScore in score.cpp
+   Segment * appendCopiesOfMeasures(Score * score,Measure * fm,Measure * lm, Segment * cur_time_sigs) {
+
+      int tickOfAppend = score->last()->endTick();
+
+      // clone measures and add them to the end of the score
+      TieMap tieMap;
+      MeasureBase * cmb = toMeasureBase(fm);
+      while (true) {
+         MeasureBase * nmb;
```

Comment:
but otherwose 6 spaces pleas, not just 4

Reply:
@Jojo-Schmitz is there a formatting description file somewhere for MuseScore? I did as much as I could by hand but the easiest way to make sure nothing was missed would be to run an automated formatter.

Reply:
Yes, there is, see https://musescore.org/en/handbook/developers-handbook/finding-your-way-around/musescore-coding-rules

Reply:
@Jojo-Schmitz ok, thank you. After reading the coding rules I see that I clearly violate a few more of them. I will try to fire up QTCreator and see how much it can fix as opposed to what still needs to be done by hand.

Reply:
for QtCreator there an xml file at the end of that handbook pages that you can download and import in QtCreator for it to obey most rules automagically

Reply:
@Jojo-Schmitz I have opened the file in QtCreator and told it to use the xml style file. It is not auto-fixing the mistakes, or even pointing them out, however. And when googling, it seems you need to download a plugin to beautify the code and the plugin uses either clang or artistic syntax, not the xml you provided.

Sorry if this is a stupid question, but I usually use another code environment so qtcreator is new for me.

Reply:
It is possible to compile MuseScore using Visual Studio if this is the one you are get used to. And XCode ofc.

Reply:
@anatoly-os no, I'm on Linux (Ubuntu 18.04 LTS). I am just used to using sublime for editing as all my other work is not in Qt. 
It may make sense to invest a bit of time to create an equivalent .clang-format file as that is more universal, being supported by most editors (including QtCreator)